integration_tests/suite/test_phonebook_http.py

```diff
@@ -358,3 +391,304 @@ def test_get_multi_tenant(self, sub, main):
+                response = client.session().get(
+                    url=f"{client.phonebook_source.base_url}/{source['uuid']}/contacts",
+                    params={'limit': 100, 'offset': i * limit},
+                )
+                response.raise_for_status()
+                body = response.json()
+                assert_that(
+                    body,
+                    has_entries(
+                        total=len(self.contacts),
+                        filtered=len(self.contacts),
+                        items=all_of(
+                            has_length(100),
+                            only_contains(
+                                has_entries(
+                                    firstname=instance_of(str),
+                                    lastname=instance_of(str),
+                                    number=instance_of(str),
+                                    id=instance_of(str),
+                                )
+                            ),
+                        ),
+                    ),
+                )
+                contact_ids.update(contact['id'] for contact in body['items'])
+
+            assert_that(
+                contact_ids, equal_to(set(contact['id'] for contact in self.contacts))
+            )
+
+    def test_get_filtered(self):
+        client = self.get_client(VALID_TOKEN_MAIN_TENANT)
+
+        with self.source(
+            client,
+            {
+                'phonebook_uuid': self.phonebook['uuid'],
+                'name': self.phonebook['name'] + "-source",
+            },
+        ) as source:
+            response = client.session().get(
+                url=f"{client.phonebook_source.base_url}/{source['uuid']}/contacts",
+                params={'search': 'Contact 4'},
+            )
+            response.raise_for_status()
+            body = response.json()
+            assert_that(
+                body,
+                has_entries(
+                    total=len(self.contacts),
+                    filtered=1111,
+                    items=all_of(
+                        has_length(1111),
+                        only_contains(
+                            has_entries(
+                                firstname=instance_of(str),
+                                lastname=instance_of(str),
+                                number=instance_of(str),
+                                id=instance_of(str),
+                            )
+                        ),
+                    ),
+                ),
+            )
+            contact_ids = set(contact['id'] for contact in body['items'])
+
+            assert_that(
+                contact_ids,
+                equal_to(
+                    set(
+                        contact['id']
+                        for contact in self.contacts
+                        if 'Contact 4' in contact.get('firstname', '')
+                    )
+                ),
+            )
+
+
+def timed(func, *args, **kwargs):
```

Comment:
Suggestion (did not try it), not really sure if I like "leaking" the context variable `timing` out of the block:
```
@contextmanager
def timed():
    result = {}
    start = time.time()
    try:
        yield result
    finally:
        end = time.time()
        result['elapsed'] = end - start

...
with timed() as timing:
    my_function('blah')
assert timing['elapsed'] < 1
```